mk/spksrc.tc.mk

```diff
@@ -121,6 +123,12 @@ tc_vars:
 	echo TC_ENV += CPPFLAGS=\"$(CPPFLAGS) $$\(ADDITIONAL_CPPFLAGS\)\" ; \
 	echo TC_ENV += CXXFLAGS=\"$(CXXFLAGS) $$\(ADDITIONAL_CXXFLAGS\)\" ; \
 	echo TC_ENV += LDFLAGS=\"$(LDFLAGS) $$\(ADDITIONAL_LDFLAGS\)\" ; \
```

Comment:
```
        = note: /usr/bin/ld: /spksrc/toolchain/syno-armv7-7.0/work/arm-unknown-linux-gnueabi/bin/../arm-unknown-linux-gnueabi/sysroot/usr/libexec/../lib/Scrt1.o: relocations in generic ELF (EM: 40)
```

I think this error is caused by the `CFLAGS`/`CXXFLAGS` and etc. When rustc build and link the build script, it wrongly uses these target compiler/linker flags.

Reply:
See also https://github.com/yoctoproject/poky/blob/e6a9aab3ef52cfcbfbad17ad9de2e142ae32654f/meta/recipes-devtools/rust/rust.inc#L179-L185

Reply:
@messense thnx a lot for spending a few cycles on this, your advices are much appreciated.

I gave that option a try and it failed with the exact same error.  I wonder, is there a way I can use a `config.toml` while building from `pip`?  I've tried various options but didn't found any that made it being handled (maybie due to pip building in a random `/tmp/*` directory).  Perhaps there is a method to pass the config file to pip using a global option of some sort?  This would ensure the environment is being set properly, always, for any given arch being built in parallel.

Reply:
> I wonder, is there a way I can use a `config.toml` while building from `pip`?

I don't think there is as of now, but most of cargo's configuration can be set by environment variable: https://doc.rust-lang.org/cargo/reference/config.html

Reply:
You can add a global `config.toml` in `CARGO_HOME` directory though.